### PR TITLE
shab: init at 1.0.0

### PR DESCRIPTION
**Pull request by zimbatm**
###### Motivation for this change

:1st_place_medal: 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/text/shab/default.nix [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{ pkgs ? import <nixpkgs> {}, ... }:
+pkgs.callPackage ./shab.nix {}
```
Comment (markuskowa) [inline pkgs/tools/text/shab/default.nix]: Why not using `callPackage` directly from the `all-packages.nix` in top-level?

File: pkgs/tools/text/shab/shab.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
+{ bash, stdenv, lib, runCommand, writeText, fetchFromGitHub }:
+let
+  version = "1.0.0";
+
+  shab = stdenv.mkDerivation {
```
Comment (markuskowa) [inline pkgs/tools/text/shab/shab.nix]: A `meta` would be nice.

File: pkgs/tools/text/shab/default.nix (hunk 1 above)
Comment (zimbatm) [reply pkgs/tools/text/shab/default.nix]: thanks for your review, fixed in 2824cd8b6f9b8e200d1187f6369fd3fe6e815ab0

Comment (zimbatm) [reply pkgs/tools/text/shab/default.nix]: I copied most of the nix code from upstream and went too fast :)

File: pkgs/tools/text/shab/shab.nix (hunk 2 above)
Comment (zimbatm) [reply pkgs/tools/text/shab/shab.nix]: good call, fixed in 2824cd8b6f9b8e200d1187f6369fd3fe6e815ab0

**Comment by zimbatm** [comment]
@markuskowa let me know if you see anything else

**Comment by markuskowa** [comment]
@zimbatm looks good now. There's just some typos left.
Out of curiosity: the `passthru`  exposes `render` and `renderText`. I guess you want to use  shab as a build tool in other derivations?

File: pkgs/tools/text/shab/default.nix [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    installCheckPhase = ''
+      [[ "$(echo 'Hello $entity' | entity=world $out/bin/shab)" == 'Hello world' ]]
+    '';
+
+    passthru = {
+      inherit render renderText;
+    };
+
+    meta = with lib; {
+      description = "the bash templating language";
```
Comment (markuskowa) [inline pkgs/tools/text/shab/default.nix]:
```suggestion
      description = "The bash templating language";
```

File: pkgs/tools/text/shab/default.nix [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      homepage = https://github.com/zimbatm/shab;
+      licenses = licenses.unlicense;
```
Comment (markuskowa) [inline pkgs/tools/text/shab/default.nix]:
```suggestion
      license = licenses.unlicense;
```

**Comment by zimbatm** [comment]
yeah, I think it might be a good replacement for `pkgs.substituteAll` in some cases.

**Pull request merged**